sdk/eventhub/azure-eventhubs/azure/eventhub/eventprocessor/sqlite3_partition_manager.py

```diff
@@ -0,0 +1,117 @@
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# -----------------------------------------------------------------------------------
+
+import time
+import uuid
+import sqlite3
+from .partition_manager import PartitionManager
+
+
+class Sqlite3PartitionManager(PartitionManager):
+    """An implementation of PartitionManager by using the sqlite3 in Python standard library.
+    Sqlite3 is a mini sql database that runs in memory or files.
+
+
+    """
+    def __init__(self, db_filename, ownership_table="ownership"):
+        """
+
+        :param db_filename: name of file that saves the sql data.
+        Sqlite3 will run in memory without a file when db_filename is ":memory:".
+        :param ownership_table: The table name of the sqlite3 database.
+        """
+        super(Sqlite3PartitionManager, self).__init__()
+        self.ownership_table = ownership_table
+        conn = sqlite3.connect(db_filename)
+        c = conn.cursor()
+        try:
+            c.execute("create table " + ownership_table +
+                      "(eventhub_name text,"
+                      "consumer_group_name text,"
+                      "instance_id text,"
+                      "partition_id text,"
+                      "owner_level integer,"
+                      "sequence_number integer,"
+                      "offset integer,"
+                      "last_modified_time integer,"
+                      "etag text)")
+        except sqlite3.OperationalError:
+            pass
+        finally:
+            c.close()
+        self.conn = conn
+
+    async def list_ownership(self, eventhub_name, consumer_group_name):
+        cursor = self.conn.cursor()
+        try:
+            cursor.execute("select "
+                                "eventhub_name, "
+                                "consumer_group_name,"
+                                "instance_id,"
+                                "partition_id,"
+                                "owner_level,"
+                                "sequence_number,"
+                                "offset,"
+                                "last_modified_time,"
+                                "etag "
+                                "from "+self.ownership_table+" where eventhub_name=? "
+                                "and consumer_group_name=?",
+                                (eventhub_name, consumer_group_name))
+            result_list = []
+            for row in cursor.fetchall():
+                d = dict()
+                d["eventhub_name"] = row[0]
+                d["consumer_group_name"] = row[1]
+                d["instance_id"] = row[2]
+                d["partition_id"] = row[3]
+                d["owner_level"] = row[4]
+                d["sequence_number"] = row[5]
+                d["offset"] = row[6]
+                d["last_modified_time"] = row[7]
+                d["etag"] = row[8]
+                result_list.append(d)
```

Comment:
Storing the names in one place up from can reduce duplication and also ensures the order is always consistent in both places:
```
names = ['eventhub_name', 'eventhub_name', ...] 
```

Then the cursor can be:
```
cursor.execute("select " + ", ".join(names) + "from...", ...)
```
and this second part can be 
```
d = dict(zip(names, row))
```

Reply:
And actually given the create table further up, the best representation to start with might be:
```
fields = [('eventhub_name', 'text'), ('consumer_group_name', 'text') ...]
```
from which the field names can be extracted, as well as the "name type" fields for the create table. 

Do agree that it's important to know the provenance of the inputs to this SQL code re: injection attacks

Reply:
This is very good python tips that make the code concise.
For SQL injection, I don't have a way to parameterize table name. Having helper function to pre-check the table name is a way.

Reply:
sql injection is prevented by checking table name